hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SContainer.scala

```diff
@@ -81,13 +81,14 @@ trait SIndexableValue extends SValue {
   override def sizeToStoreInBytes(cb: EmitCodeBuilder): SInt64Value = {
     val storageType = this.st.storageType().asInstanceOf[PContainer]
     val length = this.loadLength()
-    val totalSize = cb.newLocal[Long]("sindexableptr_size_in_bytes", storageType.elementsOffset(length).toL)
+    val totalSize = cb.newLocal[Long]("sindexableptr_size_in_bytes")
     if (this.st.elementType.containsPointers) {
       this.forEachDefined(cb) { (cb, _, element) =>
+        cb.assign(totalSize, storageType.elementsOffset(length).toL)
         cb.assign(totalSize, totalSize + element.sizeToStoreInBytes(cb).value)
```

Comment:
This is wrong, no? You add the `elementsOffset` (meaning the number of missing bytes) for each defined element of the array rather than just once. I'm a bit surprised/concerned this didn't fail any tests.

Reply:
Yeah, no idea where that came from.

This is only used in the shuffler, and only as a heuristic, to try to balance resulting partitions by total size, not just number of rows. I agree it should be tested better, but since it's not trying to be exact, it's not obvious how to do that.